### DISCUSSION TITLE
Non-dimensionalization of Pressure

### DISCUSSION THREAD

**Discussion by gauravity30** [xcompact3d/Incompact3d#242] [Q&A] [answered]
Hello !
Whether the pressure results for the simulation stored in the `pp-x.bin` dimensionalized or non-dimensionalized ? I was trying to compute lift/drag from the pressure data but the pressure seems quite low than the expected value (Order of 10^2). So, it got me thinking if the pressure is dimensionalized or non-dimensionalized ? 

Thanks,
Gaurav

**Comment by slaizet** [comment, marked as answer]
Hi,
The pressure is non-dimensionalised with 0.5 rho U_infty. You also have a dt factor (the code computes dt * p), but this is corrected in the `rescale_pressure` subroutine.
Best,
Sylvain

**Reply by gauravity30** [reply]
Thank you for your answer ! Just wanted to confirm it `0.5 rho U_infty` or `0.5 rho U_infty^2`.  So, the value stored in `pp-x.bin` is non-dimensionalized and I need to multiply by  `0.5 rho U_infty` to get the actual pressure ?

**Reply by slaizet** [reply]
yep sorry U_infty^2 :-)

**Reply by gauravity30** [reply]
Hello Sir,
Can you provide some reference for the non-dimensionalization used in xcompact3d ? I am confused about it. 

**Question:** I have plotted the instananeous pressure at one instant for flow over flat plate at Re=10000 and AoA=10 deg. Are the values of pressure here and in statistics non-dimensionalized because the values are too high for Cp distribution ? 

![image](https://github.com/xcompact3d/Incompact3d/assets/53783610/41c7bfc0-0294-4da2-ba03-9b6423f62ccb)

**Background of the problem:**  When I integrate the pressure over the surface of the flat plate and try to compute the CL and CD from it and divide it by `0.5 rho v_{infty}^2 c`. Then the values are too low as compared to the reference value. The maximum y+ on the surface of the flat plate is less than 1 for the simulation and the distribution of pressure over the flat-plate seems to be okay except for the numerical values. 

For setting the simulations in `input.i3d`, I keep the parameters `Re` as 500,  `u1` = 20 and `u2` = 20, `ra` = 0.5 for `BC-Cylinder type` flow. 

Thank you !
Gaurav

**Reply by slaizet** [reply]
Hi,
For the normalisation of the Navier-Stokes equations, any CFD book will show you how to do it. You typically normalise with a length scale, a velocity and a density. That's it. For a cylinder, you would use the diameter(=1), the free stream velocity (=1) and the density of the flow (constant, =1 as incompressible). As a result, Re=1/nu (as you would have seen in the code). I am not sure why you have u1 and u2 equal to 20, and Re=500 (especially if you want to simulate Re=10,000).
Best wishes,
Sylvain

**Reply by gauravity30** [reply]
Thank you for your response ! 
I went by the logic that if `d` is 1, `rho` is 1, then `Re = U/nu`. So for Re=10000, `U1=U2=20`, and `1/nu=500`. 

The parameters `U1` and `U2` set the maximum and minimum inlet velocity right ? So, if I have to do a simulation of Re=10000, then should I just set the `Re` parameter as 10000 irrespective of what the `U1` and `U2` parameters are ? 

Thank You ! 

Sincerely,
Gaurav

**Reply by slaizet** [reply]
Hi,
u1 and u2 are used to define the free stream velocity so no, you cannot just set up the Re without paying attention to u1 and u2. 
Once again, this is just a standard normalisation of the N-S equations.
Best,
Sylvain

**Reply by gauravity30** [reply]
Okay sir !
 Is my method of setting the simulation correct i.e. `Re =500` and `U1=U2=20` for a flow at Re=10000. If my method is correct, then I am not able to understand why the values in pressure files actually represent ? Sorry, I am quite troubled with this. I would highly appreciate it if you could help me clear this out. I went through non-dimensionalization and normalisation of the N-S equation, but it didn't help. 

Thank You !
Sincerely,
Gaurav

**Reply by slaizet** [reply]
Your method is not correct, as explained in my previous messages.

**Reply by gauravity30** [reply]
Then how should I set my simulations other than setting `Re=10000` and `U1=U2=1` ? Can you please share ?

**Reply by slaizet** [reply]
"Is my method of setting the simulation correct i.e. Re =500 and U1=U2=500 for a flow at Re=10000" --> my answer to that question was no. You need to set Re=10000, U1=U2=1, and adimentionalise all relevant quantities in the input files with the length of your immersed plate. You just need to follow the cylinder example.

**Reply by gauravity30** [reply]
Thank you sir for clearing my doubts ! I will follow the cylinder example only for my case.
